### PR TITLE
NOISSUE: Create in memory data source for benchmarks

### PR DESCRIPTION
**Pull request by lweides**

### CODE REVIEW

File: src/jmh/java/column/store/input/CreateInput.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    public static void main(final String[] args) throws IOException {
+        var name = Column.forString("name");
+        var male = Column.forBoolean("male");
+        var age = Column.forLong("age");
+        var height = Column.forDouble("height");
+        var id = Column.forId("id");
+
+        var testDir = Path.of("testDir");
+        deleteRecursively(testDir);
```
Comment (Eliasrpx) [inline src/jmh/java/column/store/input/CreateInput.java:37]: is testdir currently a placeholder ?
would rather put it into f.e.: 'src/test/resources'

File: src/jmh/java/column/store/input/InMemoryReader.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+    @Override
+    public StringByteReader of(final StringColumn column) {
+        return (StringByteReader) readers.computeIfAbsent(column, col -> {
+            var bytes = readAllBytesOf(column);
+            return new StringByteReader(bytes);
+        });
+    }
+
+    private byte[] readAllBytesOf(final Column column) {
```
Comment (Eliasrpx) [inline src/jmh/java/column/store/input/InMemoryReader.java:82]: attempt to read all Bytes into memory, could result in a OutOfMemory Exception.
-> large number of records + many colums

**Comment by Eliasrpx** [comment]
is this intended for both parquet and csv? Because currently i dont see it be suitable for csv

File: src/jmh/java/column/store/input/InMemoryReader.java (hunk 2 above)
Comment (lweides) [reply src/jmh/java/column/store/input/InMemoryReader.java:82]: Yes, but but how big will our test set realistically be? If it is too big, we can always switch to something more sophisticated.

**Comment by lweides** [comment]
Why not?

File: src/jmh/java/column/store/input/CreateInput.java (hunk 1 above)
Comment (lweides) [reply src/jmh/java/column/store/input/CreateInput.java:37]: We can change that 😄

Comment (Eliasrpx) [reply src/jmh/java/column/store/input/CreateInput.java:37]: 👍

**Comment by Eliasrpx** [comment]
the write-part is more or less fine, but the reader currently loads the data column-wise into memory, which is fine for parquet, but for csv we have to reconstruct the full row-wise records before we feed it to csv-impl

**Comment by lweides** [comment]
I will experiment with storing the data record by record, which might be faster due to better cache locality 😄

**Review by Eliasrpx — APPROVED** [review]

**Pull request merged**